### PR TITLE
feat: Set Sentry environment to production when running from main

### PR DESCRIPTION
**Pull request by armenzg**
You can see a sample `development` error [here](https://sentry.io/organizations/sentry/issues/3123211249/?environment=development&project=5324467&query=is%3Aignored&statsPeriod=1h).

### CODE REVIEW

**Comment by armenzg** [comment]
I have updated the alert to only report on production issues.

**Review by asottile-sentry — APPROVED** [review]

File: src/main.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ Sentry.init({
   ],
   release: process.env.VERSION,
   tracesSampleRate: 1.0,
+  environment:
+    process.env.GITHUB_ACTION_REF === 'main' ? 'production' : 'development',
```
Comment (billyvg) [inline src/main.ts:62]: is this an actual GHA env var? I don't see it here https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

Comment (armenzg) [reply src/main.ts:62]: I don't think it's documented:
https://github.com/getsentry/sentry/runs/6836956146?check_suite_focus=true#step:3:173

They were introduced in here:
https://github.com/actions/runner/pull/585

**Pull request merged**